app/resources/scanned_maps/scanned_map_wayfinder.rb

```diff
@@ -34,4 +34,21 @@ def members_with_parents
       member
     end
   end
+
+  # To display and save logical order correctly, this method replaces ScannedMap members that have no ScannedMap children
+  # with their corresponding geo member (TIFF) FileSet. Geo resources are unusual in that each resource has one corresponding
+  # geo member. An atlas with multiple maps, for example, will be a set of multiple ScannedMaps with each resource corresponding
+  # to a single map. ScannedMap members with ScannedMap children are analogous to a volume in a multi-volume work.
+  def logical_structure_members
+    @logical_structure_members ||= begin
+      members_with_parents.map do |member|
+        decorator = member.decorate
+        if decorator.respond_to?(:scanned_map_members) && decorator.scanned_map_members.empty?
```

Comment:
Pulling this out to a separate method would probably let you avoid the clunky `||= begin` syntax.

Reply:
Yeah, it's it a little gnarly. I'll refactor.

Reply:
Refactored to a new method, although the comment doesn't show as `outdated` for some reason.